### PR TITLE
build: enable no-unused-variable lint rule

### PR DESCRIPTION
**Pull request by DevVersion**
* Now TSLint starts reporting unused variables again (Rule has been un-deprecated and now uses the type checker
* In favor of linting the whole project the tsconfig file inside of `src/` needs to live in the project root.

### CODE REVIEW

File: src/lib/core/datetime/native-date-adapter.spec.ts [hunk 1]
```diff
@@ -1,16 +1,9 @@
 import {NativeDateAdapter} from './native-date-adapter';
 import {Platform} from '../platform/index';
-
+import {DEC, FEB, JAN, MAR} from '../testing/date-shortcuts';
```
Comment (jelbourn) [inline src/lib/core/datetime/native-date-adapter.spec.ts]: `/testing/month-constants`?

**Comment by mmalerba** [comment]
please rebase

**Comment by DevVersion** [comment]
@mmalerba Done.

**Comment by mmalerba** [comment]
sorry, needs more rebase :p

**Comment by DevVersion** [comment]
Done 😄

**Pull request merged**